src/components/MemberList.tsx

```diff
@@ -1,39 +1,217 @@
+                <Tooltip label="LinkedIn Profile">
+                  <IconButton
+                    as="a"
+                    href={`https://linkedin.com/in/${member.linkedin.username}`}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    aria-label="LinkedIn Profile"
+                    icon={<Linkedin size={20} />}
+                    variant="ghost"
+                    color={iconColor}
+                    size="sm"
+                  />
+                </Tooltip>
+              )}
+
+              {member.leetcode?.username && !member.leetcode?.isPrivate && (
+                <Tooltip label="LeetCode Profile">
+                  <IconButton
+                    as="a"
+                    href={`https://leetcode.com/${member.leetcode.username}`}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    aria-label="LeetCode Profile"
+                    icon={<Code2 size={20} />}
+                    variant="ghost"
+                    color={iconColor}
+                    size="sm"
+                  />
+                </Tooltip>
+              )}
+            </HStack>
+          </Flex>
+
+          <Box mt={4}>
+            <Link to={`/directory/${member.id}`}>
+              <Button size="sm" colorScheme="brand">
+                View Profile
+              </Button>
+            </Link>
+          </Box>
+        </Box>
+      </HStack>
+    </Box>
+  );
+};
+
+const LoadingCard = () => {
+  const borderColor = useColorModeValue('gray.200', 'gray.700');
+
   return (
-    <VStack spacing={4} align="stretch">
-      {members.map((member) => (
-        <Box
-          key={member.id}
-          p={4}
-          borderWidth={1}
-          borderRadius="lg"
-          overflow="hidden"
-          boxShadow="sm"
-        >
-          <HStack spacing={4}>
-            <Avatar name={resolveName(member)} src={member.profilePictureUrl} />
-            <VStack align="start">
-              <Text fontWeight="bold">{member.username}</Text>
-              <Text>{resolveName(member)}</Text>
-              <Link to={`/directory/${member.id}`}>
-                <Button size="sm" colorScheme="brand">
-                  View Profile
-                </Button>
-              </Link>
-            </VStack>
-          </HStack>
+    <Box
+      p={6}
+      borderWidth={1}
+      borderRadius="lg"
+      borderColor={borderColor}
+      minH="140px"
+      w="100%"
+    >
+      <HStack spacing={4} align="start">
+        <SkeletonCircle size="16" />
+        <Box flex="1">
+          <Skeleton height="24px" width="200px" mb={2} />
+          <Skeleton height="20px" width="150px" mb={4} />
+          <Skeleton height="32px" width="100px" />
         </Box>
-      ))}
+      </HStack>
+    </Box>
+  );
+};
+
+const MemberList: React.FC<MemberListProps> = ({
+  members,
+  loading = false,
+  currentPage = 1,
```

Comment:
MaGiC nUmBeRs. I'm taking off style points @deeksha-vatwani we need you in here.

Reply:
`const ONE = 1`